### PR TITLE
Validate GitHub issue templates

### PR DESCRIPTION
**Pull request by dstansby**
This validates the GitHub issue tempaltes against a schema, and would have caught the issue that was fixed by https://github.com/UCL-ARC/python-tooling/pull/306. 

Unfortunately it's not possible to validate the package template files as they contain invalid characters for a string entry (`{` and `}`).

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -56,3 +56,12 @@ repos:
...
       - id: trailing-whitespace
+  - repo: https://github.com/python-jsonschema/check-jsonschema
+    rev: cb934cfd6d2eea1aed7b20e4e9e6339d5952322c # 0.28.0
+    hooks:
+      # Schemas taken from https://www.schemastore.org/json/
+      - id: check-jsonschema
+        name: "Validate GitHub issue templates"
+        files: ^\.github/ISSUE_TEMPLATE/.*\.yml$
+        exclude: ^\.github/ISSUE_TEMPLATE/config\.yml$
+        args: ["--verbose", "--schemafile", "schemas/github-issue-forms.json"]
```
Comment (paddyroddy) [inline .pre-commit-config.yaml]: Why can you add this to the `.pre-commit-config.yaml` of the template? Then they both get validation

File: schemas/github-issue-forms.json [hunk 2]
```diff
@@ -0,0 +1,2377 @@
+{
```
Comment (paddyroddy) [inline schemas/github-issue-forms.json:1]: Is it necessary to add this to the repo? Like is there a way to do this remotely?

**Review by paddyroddy — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (dstansby) [reply .pre-commit-config.yaml]: > Unfortunately it's not possible to validate the package template files as they contain invalid characters for a string entry ({ and }).

File: schemas/github-issue-forms.json (hunk 2 above)
Comment (dstansby) [reply schemas/github-issue-forms.json:1]: I don't think there's another way

File: .pre-commit-config.yaml (hunk 1 above)
Comment (dstansby) [reply .pre-commit-config.yaml]: Ah sorry, just realised what you meant! Good shout, I will do that.

Comment (dstansby) [reply .pre-commit-config.yaml]: 👍 done

Comment (paddyroddy) [reply .pre-commit-config.yaml]: I did wonder if you'd missed something

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -56,3 +56,12 @@ repos:
         args:
           - --fix=lf
       - id: trailing-whitespace
+  - repo: https://github.com/python-jsonschema/check-jsonschema
+    rev: cb934cfd6d2eea1aed7b20e4e9e6339d5952322c # 0.28.0
```
Comment (paddyroddy) [inline .pre-commit-config.yaml]: I assume `renovate` will use this syntax in the future. How does this play with `pre-commit autoupdate`?

**Review by paddyroddy — COMMENTED** [review]
Sorry thought of one more thing

File: .pre-commit-config.yaml (hunk 3 above)
Comment (dstansby) [reply .pre-commit-config.yaml]: 😢 `pre-commit autoupdate` doesn't like it, and just changes it back to a version number. So I've switched the SHAs to version numbers.

Comment (paddyroddy) [reply .pre-commit-config.yaml]: Yeah, I thought that might be the case. The `pre-commit` maintainer is pretty stubborn, too, so he likely won't change the format.

**Review by paddyroddy — APPROVED** [review]

**Pull request merged**